libs/application/template-api-modules/src/lib/modules/shared/services/attachment-s3.service.ts

```diff
@@ -25,22 +25,16 @@ export class AttachmentS3Service {
   ): Promise<AttachmentData[]> {
     const attachments: AttachmentData[] = []
 
-    for (let i = 0; i < attachmentAnswerKeys.length; i++) {
-      const answers = getValueViaPath(
-        application.answers,
-        attachmentAnswerKeys[i],
-      ) as Array<{
+    attachmentAnswerKeys.forEach(async (key) => {
+      const answers = getValueViaPath(application.answers, key) as Array<{
         key: string
         name: string
       }>
-      if (!answers) continue
-      const list = await this.toDocumentDataList(
-        answers,
-        attachmentAnswerKeys[i],
-        application,
-      )
+      if (!answers) return
+      const list = await this.toDocumentDataList(answers, key, application)
```

Comment:
Actually, on second look I see that you are awaiting a promise inside the loop, did you test this after changing it? I'm 90% this won't work correctly in a forEach, instead you should use `for (const key of attachmentAnswerKeys)` so that it correctly awaits response in each loop

https://stackoverflow.com/questions/37576685/using-async-await-with-a-foreach-loop

Reply:
I can only test attachments on feature deploy so I haven't done it yet, but will take one round of testing once the build is done.